### ISSUE TITLE
match is a valid method name

### ISSUE THREAD

**Issue by tstarling** [microsoft/tolerant-php-parser#392]
`match` can still be used as a method name and as a class constant name in PHP 8.0, but tolerant-php-parser gives an error.

```
$parser = new Microsoft\PhpParser\Parser;
$ast = $parser->parseSourceFile('<?php class C { function match() {} }');
var_dump(Microsoft\PhpParser\DiagnosticsProvider::getDiagnostics($ast));
```

```
array(7) {
  [0]=>
  object(Microsoft\PhpParser\Diagnostic)#480 (4) {
    ["kind"]=>
    int(0)
    ["message"]=>
    string(16) "'Name' expected."
    ["start"]=>
    int(24)
    ["length"]=>
    int(0)
  }
...
```

**Comment by Krinkle** [comment]
From https://github.com/microsoft/tolerant-php-parser/issues/365 and https://github.com/microsoft/tolerant-php-parser/pull/366:
> Needs to be added in TokenStringMaps under KEYWORDS array.
> https://www.php.net/manual/en/reserved.keywords.php

What that page actually says:

> The following words cannot be used as [global] constants, class names, or [global] function names. They are, however, **allowed as property, constant, and method names of classes**, interfaces and traits, […].

**Comment by Krinkle** [comment]
Hm.. I misinterpeted the PR at https://github.com/microsoft/tolerant-php-parser/pull/366. Whilst preparing a revert with test case locally, I realized the test was failing and passing in the opposite direction as I expected. Turns out PR https://github.com/microsoft/tolerant-php-parser/pull/366 isn't causing `match` to be seen as a match operator, rather it is merely turning the token from known to unknown. tolerant-php-parser already knew to allow "keyword" tokens as legal method names, so all is well.

From https://github.com/phan/phan/issues/4751:

> It seems the uptream issue at https://github.com/microsoft/tolerant-php-parser/issues/392 was already fixed back in June 2022, before I reported this issue here issue. The fix was released in tolerant-php-parser v0.1.2 on 5 Oct 2022, and Phan's main branch was updated the same day, https://github.com/phan/phan/commit/cc66093646875525ea23a4f72ebb82dd0a5081d6.
> 
> However, it remained unreleased for several months, until the Phan 5.4.2 release in March 2023. https://github.com/phan/phan/releases/tag/5.4.2
> 
> The example code now analyzes without error <https://phan.github.io/demo/>. Closing as such. Thanks!

I believe this report can likewise we closed a well.
